crates/app-data/src/app_data.rs

```diff
@@ -239,6 +242,79 @@ impl<'de> Deserialize<'de> for OrderUid {
     }
 }
 
+/// A list containing all the partner fees
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+#[cfg_attr(
+    any(test, feature = "test_helpers"),
+    derive(Serialize),
+    serde(transparent)
+)]
+pub struct PartnerFees(Vec<PartnerFee>);
+
+impl PartnerFees {
+    pub fn iter(&self) -> Iter<'_, PartnerFee> {
+        self.0.iter()
+    }
+}
+
+impl<'de> Deserialize<'de> for PartnerFees {
+    fn deserialize<D>(deserializer: D) -> Result<PartnerFees, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        PartnerFeeHelper::deserialize(deserializer).map(|v| v.into_partner_fees())
+    }
+}
+
+enum PartnerFeeHelper {
+    Empty,
```

Comment:
I think you don't need the `Empty` case if you keep the `partner_fee` as optional. If the field does not exist in the JSON the custom deserialization function will not be called.